pytest_girder/pytest_girder/utils.py

```diff
@@ -0,0 +1,143 @@
+                      of the form 'login:password'
+    :param exception: Set this to True if a 500 is expected from this call.
+    :param cookie: A custom cookie value to set.
+    :param token: If you want to use an existing token to login, pass
+        the token ID.
+    :type token: str
+    :param additionalHeaders: A list of headers to add to the
+                              request.  Each item is a tuple of the form
+                              (header-name, header-value).
+    :param useHttps: If True, pretend to use HTTPS.
+    :param authHeader: The HTTP request header to use for authentication.
+    :type authHeader: str
+    :param appPrefix: The CherryPy application prefix (mounted location without trailing slash)
+    :type appPrefix: str
+    :returns: The cherrypy response object from the request.
+    """
+    local = cherrypy.lib.httputil.Host('127.0.0.1', 30000)
+    remote = cherrypy.lib.httputil.Host('127.0.0.1', 30001)
+    headers = [('Host', '127.0.0.1'), ('Accept', 'application/json')]
+    qs = fd = None
+
+    if additionalHeaders:
+        headers.extend(additionalHeaders)
+
+    if isinstance(body, six.text_type):
+        body = body.encode('utf8')
+
+    if params:
+        qs = urllib.parse.urlencode(params)
+
+    if params and body:
+        # In this case, we are forced to send params in query string
+        fd = BytesIO(body)
+        headers.append(('Content-Type', type))
+        headers.append(('Content-Length', '%d' % len(body)))
+    elif method in ['POST', 'PUT', 'PATCH'] or body:
+        if type:
+            qs = body
+        elif params:
+            qs = qs.encode('utf8')
+
+        headers.append(('Content-Type', type or 'application/x-www-form-urlencoded'))
+        headers.append(('Content-Length', '%d' % len(qs or b'')))
+        fd = BytesIO(qs or b'')
+        qs = None
+
+    app = cherrypy.tree.apps[appPrefix]
+    request, response = app.get_serving(
+        local, remote, 'http' if not useHttps else 'https', 'HTTP/1.1')
+    request.show_tracebacks = True
+
+    headers = buildHeaders(headers, cookie, user, token, basicAuth, authHeader)
+
+    # Python2 will not match Unicode URLs
+    url = str(prefix + path)
+    try:
+        response = request.run(method, url, qs, 'HTTP/1.1', headers, fd)
+    finally:
+        if fd:
+            fd.close()
+
+    if isJson:
+        body = getResponseBody(response)
+        try:
+            response.json = json.loads(body)
+        except Exception:
+            print(url)
+            print(body)
+            raise AssertionError('Did not receive JSON response')
+
+    if not exception and response.output_status.startswith(b'500'):
+        raise AssertionError("Internal server error: %s" %
+                             getResponseBody(response))
+
+    return response
+
+
+def buildHeaders(headers, cookie, user, token, basicAuth, authHeader):
+    from girder.models.token import Token
```

Comment:
@jbeezley I understood your commit moving imports into the fixture functions since fixtures.py is automatically imported, but shouldn't we be able to import this at the top level since test modules have to import it explicitly?

Reply:
No, it is still necessary because the utils module is imported by the fixtures module.